### ISSUE TITLE
sc-im: "Illegal hardware instruction" when trying to save

### ISSUE THREAD

**Issue by Nitroretro** [termux/termux-packages#4614]
**Problem description**
When I try to save, sc-im crashes and the shell displays "zsh: Illegal hardware instruction". The file doesn't actually get created.

**Steps to reproduce**
1. `$ scim`
2. `:w Example`

**Expected behavior**
sc-im should save the file without crashing.

**Additional information**
```shell
$ termux-info
Packages CPU architecture:
arm
Subscribed repositories:
# sources.list
deb https://termux.org/packages/ stable main
# science-repo (sources.list.d/science.list)
deb https://dl.bintray.com/grimler/science-packages-24 science stable
# game-repo (sources.list.d/game.list)
deb https://dl.bintray.com/grimler/game-packages-24 games stable
Updatable packages:
All packages up to date
Android version:
7.1.1
Kernel build information:
Linux localhost 3.10.49-14895767 #1 SMP PREEMPT Wed Sep 11 14:54:01 KST 2019 armv7l Android
Device manufacturer:
samsung
Device model:
SM-A5100
$ logcat -d "*W" | grep scim
12-02 08:57:22.730 11930 11930 F libc    : Fatal signal 4 (SIGILL), code 1, fault addr 0xb336e9d0 in tid 11930 (scim)
```

**Comment by xeffyr** [comment]
What's your output of `cat /proc/cpuinfo` ?

**Comment by Nitroretro** [comment]
```shell
$ cat /proc/cpuinfo
processor	: 0
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 38.40
Features	: swp half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 evtstrm 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 1
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 38.40
Features	: swp half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 evtstrm 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 2
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 38.40
Features	: swp half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 evtstrm 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 3
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 38.40
Features	: swp half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 evtstrm 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 4
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 38.40
Features	: swp half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 evtstrm 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 5
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 38.40
Features	: swp half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 evtstrm 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 6
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 38.40
Features	: swp half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 evtstrm 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 7
model name	: ARMv7 Processor rev 4 (v7l)
BogoMIPS	: 38.40
Features	: swp half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 evtstrm 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

Hardware	: Qualcomm Technologies, Inc MSM8939
Revision	: 0002
Serial		: 0000b8580000ecef
Processor	: ARMv7 Processor rev 4 (v7l)
```

**Comment by Nitroretro** [comment]
This seems to be related: andmarti1424/sc-im#364

**Comment by xeffyr** [comment]
Confirming - it crashes on all my devices (independently of arch & Android versions).

It looks like problem is not related to opcodes supported by CPU as in https://github.com/termux/termux-packages/issues/4614#issuecomment-560368251 all expected features are present.

`sc-im` does not crash under gdb but freezes.

**Comment by its-pointless** [comment]
Is this still an issue cause i can get an implementation of wordexp working?

**Comment by xeffyr** [comment]
@its-pointless Could you submit patches ?

**Comment by xeffyr** [comment]
Fixed in `0.7.0-5`.
